python/beeai_framework/retryable.py

```diff
@@ -0,0 +1,247 @@
+# Copyright 2025 IBM Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import asyncio
+import uuid
+from collections.abc import AsyncGenerator, Awaitable, Callable
+from enum import Enum
+from typing import Any, Self, TypeVar
+
+from pydantic import BaseModel
+
+from beeai_framework.cancellation import AbortController, AbortSignal, signal_race
+from beeai_framework.errors import FrameworkError
+from beeai_framework.utils.custom_logger import BeeLogger
+from beeai_framework.utils.models import ModelLike, to_model
+
+T = TypeVar("T", bound=BaseModel)
+logger = BeeLogger(__name__)
+
+
+class TaskState(str, Enum):
+    PENDING: str = "PENDING"
+    RESOLVED: str = "RESOLVED"
+    REJECTED: str = "REJECTED"
+
+    def __str__(self) -> str:
+        return self.value
+
+
+class Task:
```

Comment:
This is JS Promise Wrapper; this should not be needed. 
Please remove.

Reply:
https://github.com/i-am-bee/beeai-framework/pull/363/commits/99e2ed0675b66a415cfc229984ff812d5457d4ed